api/hasura/actions/_handlers/vouch.ts

```diff
@@ -18,25 +19,56 @@ import {
 import { Awaited } from '../../../../api-lib/ts4.5shim';
 import { verifyHasuraRequestMiddleware } from '../../../../api-lib/validate';
 import {
+  composeHasuraActionRequestBodyWithApiPermissions,
   composeHasuraActionRequestBodyWithSession,
   HasuraUserSessionVariables,
+  vouchApiInput,
   vouchInput,
 } from '../../../../src/lib/zod';
 
 type Voucher = Awaited<ReturnType<typeof getUserFromProfileId>>;
 type Nominee = Awaited<ReturnType<typeof getNominee>>;
 
+const userRequestSchema = composeHasuraActionRequestBodyWithSession(
+  vouchInput,
+  HasuraUserSessionVariables
+);
+
+const apiRequestSchema = composeHasuraActionRequestBodyWithApiPermissions(
+  vouchApiInput,
+  ['create_vouches']
+);
+
+const requestSchema = z.union([userRequestSchema, apiRequestSchema]);
+
 async function handler(req: VercelRequest, res: VercelResponse) {
   try {
     const {
       input: { payload: input },
       session_variables: sessionVariables,
-    } = composeHasuraActionRequestBodyWithSession(
-      vouchInput,
-      HasuraUserSessionVariables
-    ).parse(req.body);
+    } = await requestSchema.parse(req.body);
+
+    let voucherProfileId: number | null;
+    // Allow passing in voucher_user_id for api-users to vouch on behalf of a user
+    if (sessionVariables.hasuraRole === 'api-user') {
+      assert('voucher_user_id' in input, 'voucher_user_id not specified');
+      voucherProfileId = input.voucher_user_id;
```

Comment:
why is this a profile ID instead of a user ID? the field is called user_id in the input

Reply:
Not sure tbh, it's copy pasted to handle the api-user role. Can take a deeper look if you want me to address in this PR

Reply:
renaming to `voucher_profile_id`

Reply:
I don't see this change pushed?